python/tests/stress/arcticdb/version_store/test_mem_leaks.py

```diff
@@ -146,13 +154,132 @@ def grow_exp(df_to_grow: pd.DataFrame, num_times_xx2: int):
         df_to_grow = pd.concat([df_to_grow, df_prev])
     return df_to_grow
 
+def generate_big_dataframe(rows:int=1000000):
+    print("Generating big dataframe")
+    st = time.time()
+    df = get_sample_dataframe(rows)
+    df = grow_exp(df, 5)
+    print("Generation took :", time.time() - st)
+    return df
+
+#endregion
+
+#region HELPER functions for memray tests
+
+def construct_df_querybuilder_tests(size: int) -> pd.DataFrame:
+    df = get_sample_dataframe(size)
+    date_range = pd.date_range(start="2000-1-1", periods=size, freq="s")
+    df.index = pd.Index(date_range)
+    df.index.name = "timestamp"
+    return df
+
+
+def query_group() -> QueryBuilder:
+    """
+        groupby composite aggregation query for QueryBuilder memory tests.
+        The query basically will do aggregation of half of dataframe
+    """
+    q = QueryBuilder()
+    return (
+        q[q["bool"] == True]
+        .groupby("uint8")
+        .agg({"uint32": "mean", "int32": "sum", "strings": "count", "float64": "sum", "float32": "min", "int16": "max"})
+    )
+
+
+def query_group_bigger_result() -> QueryBuilder:
+    """
+        groupby composite aggregation query for QueryBuilder memory tests.
+        The query basically will do aggregation of whole dataframe
+    """
+    q = QueryBuilder()
+    return (
+        q[q["strings"] != "QASDFGH"]
+        .groupby("int16")
+        .agg({"uint32": "mean", 
+              "int32": "sum", 
+              "strings": "count", 
+              "float64": "sum", 
+              "float32": "min", 
+              "int16": "max"})
+    )
+
+
+def query_apply(str:str) -> QueryBuilder:
+    """
+        Apply query for QueryBuilder memory tests.
+        This version of apply does couple of nested operations
+        with columns that were just added
+    """
+    q = QueryBuilder()
+    q = q[q["strings"] != str].apply("NEW 1", q["uint32"] * q["int32"] / q["float32"])
+    q = q.apply("NEW 2", q["float32"] + q["float64"] + q["NEW 1"])
+    q = q.apply("NEW 3", q["NEW 2"] - q["NEW 1"] + q["timestamp"])
```

Comment:
It is a bug that this works, we shouldn't allow adding timestamp columns to numeric columns

Reply:
Then that is tricky thing - if that is released it is a feature :-) especially if someone lik me has found that it works. So now we could have potentially lots of usages (or none)

Anyway I can and will change that, but perhaps I should create an issue on that to discuss this ?

Reply:
Sure, add a low-priority bug to Monday

Reply:
https://man312219.monday.com/boards/7852509418/pulses/8065794446